### PR TITLE
Add typer

### PR DESCRIPTION
**Pull request by dfornika**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dfornika** [comment]
@conda-forge/help-python I think this is ready to review

File: recipes/typer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - typer
+
+about:
+  home: https://github.com/tiangolo/typer
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
+  summary: 'A library for building CLI applications '
+
+  # The remaining entries in this section are optional, but recommended.
```
Comment (synapticarbors) [inline recipes/typer/meta.yaml:40]: delete comment

File: recipes/typer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: True  # [py<36]
```
Comment (synapticarbors) [inline recipes/typer/meta.yaml:16]: conda-forge is no longer building packages for py<3.6, so I think we can consider dropping the `skip` and seeing if we can make this `noarch: python`:

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

if you want to make this explicit, you can change `python` to `python >=3.6` under both `host` and `run`

File: recipes/typer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+    - flit-core
+  run:
+    - python
+    - click
+    - colorama
```
Comment (synapticarbors) [inline recipes/typer/meta.yaml:26]: It looks like `colorama` and `shellingham` are optional dependencies, but I guess that's ok to make this the equivalent of `pip install typer[all]`

**Pull request merged**